File: algorithms/TopKFrequent.java
package algorithms;

import java.util.*;

public class TopKFrequent {
    public static int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> frequencyMap = new HashMap<>();
        for (int num : nums) {
            frequencyMap.put(num, frequencyMap.getOrDefault(num, 0) + 1);
        }
        List<Integer> topK = frequencyMap.entrySet()
                .stream()
                .sorted((a, b) -> b.getValue() - a.getValue())
                .limit(k)
                .map(Map.Entry::getKey)
                .toList();
        int[] result = new int[k];
        for (int i = 0; i < k; i++) {
            result[i] = topK.get(i);
        }
        return result;
    }

    public static void main(String[] args) {
        int[] nums = { 1, 1, 1, 2, 2, 3, 3, 3 };
        int k = 2;
        int[] topK = topKFrequent(nums, k);
        System.out
                .println("Top " + k + " frequent elements in " + Arrays.toString(nums) + ": " + Arrays.toString(topK));
    }
}

File: algorithmsJS/TopKFrequent.js
class TopKFrequent {
  static topKFrequent(nums, k) {
    const frequencyMap = new Map();
    for (let num of nums) {
      frequencyMap.set(num, (frequencyMap.get(num) || 0) + 1);
    }
    const topK = Array.from(frequencyMap.entries())
      .sort((a, b) => b[1] - a[1])
      .slice(0, k)
      .map((entry) => entry[0]);
    return topK;
  }

  static topKFrequent1(nums, k) {
    return null;
  }
}

// Example usage:
const nums = [1, 1, 1, 2, 2, 3, 3, 3];
const k = 2;
const topK = TopKFrequent.topKFrequent(nums, k);
const topK1 = TopKFrequent.topKFrequent1(nums, k);
console.log(`Top ${k} frequent elements in ${JSON.stringify(nums)}:`, topK);
console.log(`Top ${k} frequent elements in ${JSON.stringify(nums)}:`, topK1);
